botocore/utils.py

```diff
@@ -1254,37 +1262,30 @@ def is_valid_uri(endpoint_url):
     )
 
 
-def validate_region_name(region_name):
-    """Provided region_name must be a valid host label."""
-    if region_name is None:
-        return
-    valid_host_label = re.compile(r'^(?![0-9]+$)(?!-)[a-zA-Z0-9-]{,63}(?<!-)$')
-    valid = valid_host_label.match(region_name)
-    if not valid:
-        raise InvalidRegionError(region_name=region_name)
```

Comment:
The pattern in this function is different from the ones used for validating bucket names. And that makes sense because the rules for region names are different from bucket names. One difference is that bucket names can be all-digits but region names cannot.

Reply:
ah yeah i missed the subtle differences and the variable name made me think it was the same for some reason. I can put it back